mythril_core/src/vga_logger.rs

```diff
@@ -0,0 +1,101 @@
+use core::fmt;
+
+const VGA_BASE_ADDR: usize = 0xB8000;
+const VGA_WIDTH: usize = 80;
+const VGA_HEIGHT: usize = 25;
+const VGA_ATTRIB: u16 = 0x0F00; // black background, white text
+
+fn get_vga_addr(x: usize, y: usize) -> usize {
```

Comment:
Rather than doing this, I think we should probably have the writer store a `&[[u16; VGA_WIDTH]; VGA_HEIGHT]` (I may have that backwards). And index in to that. Then we just need to do one unsafe conversion from `0xB8000` to that array ref, rather than using raw pointers everywhere.

Reply:
I wasn't able to create a direct reference in the `VgaWriter` struct itself due to this issue: https://github.com/rust-lang/rust/issues/57349

Instead, i've made it so that the only raw pointer deref occurs in the `write` method which then passes the reference to the vga functions.